kornia/testing/__init__.py

```diff
@@ -76,30 +76,44 @@ def create_random_fundamental_matrix(batch_size, std_val=1e-3):
 
 
 class BaseTester(ABC):
+    DTYPE_PRECISIONS = {torch.float16: (1e-2, 1e-3), torch.float32: (1e-4, 1e-4), torch.float64: (1e-5, 1e-8)}
+
     @abstractmethod
-    def test_smoke(self):
+    def test_smoke(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
     @abstractmethod
-    def test_exception(self):
+    def test_exception(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
     @abstractmethod
-    def test_cardinality(self):
+    def test_cardinality(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
     @abstractmethod
-    def test_jit(self):
+    def test_jit(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
     @abstractmethod
-    def test_gradcheck(self):
+    def test_gradcheck(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
     @abstractmethod
-    def test_module(self):
+    def test_module(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
+    def assert_close(self, actual: torch.Tensor, expected: torch.Tensor, low_tolerance: bool = False) -> None:
```

Comment:
```suggestion
    def assert_close(self, actual: Tensor, expected: Tensor, low_tolerance: bool = False) -> None:
```
`from kornia.core import Tensor`